clients/application/src/main/java/org/eclipse/hono/application/client/DownstreamMessage.java

```diff
@@ -123,4 +125,30 @@ default boolean isSenderConnected(final Instant now) {
         }
     }
 
+    /**
+     * Returns the time until disconnection notification of this downstream message.
+     *
+     * @return A notification if the message contains a TTD value {@link Optional#empty()} otherwise.
+     */
+    default Optional<TimeUntilDisconnectNotification> getTimeUntilDisconnectNotification() {
+
+        final Integer ttd = getTimeTillDisconnect();
+        final Instant creationTime = getCreationTime();
+
+        if (ttd == null) {
+            return Optional.empty();
+        } else if (ttd == 0 || MessageHelper.isDeviceCurrentlyConnected(ttd,
```

Comment:
I'd rather not introduce this dependency on *MessgeHelper* here as it is basically a helper class for dealing with AMQP messages and pulls in a lot of transitive dependencies. Maybe we can encapsulate the required logic in *TimeUntilDisconnectNotification* as well?

Reply:
Done.